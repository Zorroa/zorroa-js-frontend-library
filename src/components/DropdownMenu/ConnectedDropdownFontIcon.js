import { connect } from 'react-redux'
import DropdownFontIcon from './DropdownFontIcon'

export default connect(state => ({
  dark: state.app.monochrome,
}))(DropdownFontIcon)
