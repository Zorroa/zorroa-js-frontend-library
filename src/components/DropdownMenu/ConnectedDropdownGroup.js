import { connect } from 'react-redux'
import DropdownGroup from './DropdownGroup'

export default connect(state => ({
  dark: state.app.monochrome,
}))(DropdownGroup)
