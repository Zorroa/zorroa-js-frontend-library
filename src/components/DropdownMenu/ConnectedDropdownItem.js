import { connect } from 'react-redux'
import DropdownItem from './DropdownItem'

export default connect(state => ({
  dark: state.app.monochrome,
}))(DropdownItem)
