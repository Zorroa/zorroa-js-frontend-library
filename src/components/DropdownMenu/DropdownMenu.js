import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import classnames from 'classnames'

class DropdownMenu extends PureComponent {
  static propTypes = {
    dark: PropTypes.bool.isRequired,
    children: React.PropTypes.any,
  }

  isDark() {
    return this.props.dark === true
  }

  render() {
    const dropdownItemCLasses = classnames('DropdownMenu', {
      'DropdownMenu--dark': this.isDark(),
    })
    return <div className={dropdownItemCLasses}>{this.props.children}</div>
  }
}

export default DropdownMenu
