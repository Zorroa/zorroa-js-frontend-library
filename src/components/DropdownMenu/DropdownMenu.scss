@import '../../styles/core-mixins.scss';
@import '../../styles/colors.scss';

.DropdownMenu {
  background: $zorroa-color-white;
  box-shadow: 0 4px 6px rgba($zorroa-color-black, 0.2);

  &--dark {
    background: $zorroa-color-black;
    box-shadow: 0 4px 6px rgba($zorroa-color-black, 0.9);
  }

  &__group {
    padding: $zorroa-metric-base / 2 0;
    border: 0 solid $zorroa-color-gray-light-2;

    &--dark {
      border-color: $zorroa-color-gray-dark-2;
    }

    & ~ .DropdownMenu__group {
      border-top-width: 1px;
    }
  }

  &__font-icon {
    margin-right: $zorroa-metric-base;
  }

  &__item {
    padding: $zorroa-metric-base $zorroa-metric-base * 2;
    white-space: nowrap;
    color: $zorroa-color-gray-dark-2;
    cursor: pointer;
    display: flex;
    flex-direction: row;
    align-items: center;
    align-content: center;

    &:hover {
      background-color: $zorroa-color-gray-light-2;
    }

    &--dark {
      color: $zorroa-color-gray-light-2;

      &:hover {
        background-color: $zorroa-color-gray-dark-2;
      }
    }

    &--disabled {
      opacity: .5;
      cursor: not-allowed;
      &:hover {
        background-color: transparent;
      }
    }
  }
}
