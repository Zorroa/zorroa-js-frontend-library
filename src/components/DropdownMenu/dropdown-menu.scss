.dropdown-menu {
  button {

  }

  ul {
    background-clip: padding-box;
    background-color: #fff;
    border-radius: 4px;
    border: 1px solid #ccc;
    font-size: 14px
    list-style-type: none;
    margin: 0;
    min-width: 160px;
    padding: 15px 18px;
    position: absolute;
    text-align: left;
    top: 100%;
    z-index: 1000;
  }

  li {
    line-height: 1.5
  }

  .dropdown-caret {
    display: inline-block;
    width: 0;
    height: 0;
    margin-left: 2px;
    vertical-align: middle;
    border-top: 4px solid;
    border-right: 4px solid transparent;
    border-left: 4px solid transparent;
  }
}
