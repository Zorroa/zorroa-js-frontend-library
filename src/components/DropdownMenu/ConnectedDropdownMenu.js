import { connect } from 'react-redux'
import DropdownMenu from './DropdownMenu'

export default connect(state => ({
  dark: state.app.monochrome,
}))(DropdownMenu)
