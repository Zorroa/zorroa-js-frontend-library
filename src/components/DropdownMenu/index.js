import './DropdownMenu.scss'
import DropdownMenu from './ConnectedDropdownMenu'
import DropdownItem from './ConnectedDropdownItem'
import DropdownGroup from './ConnectedDropdownGroup'
import DropdownFontIcon from './DropdownFontIcon'

export default DropdownMenu
export { DropdownGroup, DropdownItem, DropdownFontIcon }
