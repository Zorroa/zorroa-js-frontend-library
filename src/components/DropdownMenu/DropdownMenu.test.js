/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import DropdownMenu from './DropdownMenu'

describe('<DropdownMenu />', () => {
  it('It should render the proper markup', () => {
    const tree = renderer.create(<DropdownMenu dark={false} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
