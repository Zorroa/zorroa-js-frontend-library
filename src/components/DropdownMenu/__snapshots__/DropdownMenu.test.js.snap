exports[`<DropdownMenu /> It should render the proper markup 1`] = `
<div
  className="DropdownMenu" />
`;
